standard/docs/en/schema/merging.md

```diff
@@ -51,106 +52,163 @@ If the record is complete with versioning information, then the versioning secti
+1. Get all releases with the same `ocid` value
+1. Order the releases in chronological order by `date`
+1. Create an empty JSON object for the compiled or versioned release
+1. Merge each release (**input**), in order, into the JSON object (**output**), as follows:
 
-In a versioned release, instead of over-writing past values when combining multiple releases, each field (with the exception of the `id` property of objects within an array) becomes an array of objects, indicating the:
+#### Object values
 
-* The date, id and tag of the releases where a field-value pair was first encountered;
-* The value of the field-value pair at that point;
+The release is itself an object, so this case is encountered first.
 
-As a result, the history of any field can be easily read from the data structure.
+If the object is empty in **input**, do nothing. For each field within the object in **input**:
 
-The property `"versionId":true` is used to explicitly declare the cases where an `id` field **should** be versioned (i.e. within an object that is not within an array). 
+* For a compiled release:
+  * If the field in **input** has a value of `null`, remove the field from the object in **output**, if present
+  * If the field isn't in **output**, add the field to the object in **output**, and set it to its value in **input**
+  * If the field is in **output**, merge the field's values in **output** and **input** according to the merge routine
+* For a versioned release:
+  * Merge the field's values in **output** and **input** according to the merge routine; if there is a result, add the field to the object in **output** if not already added, and set it to the result
 
-### Example
+#### Literal values
+
+If the **input** value is neither an object nor an array, then:
+
+* For a compiled release:
+  * If the **input** value is not `null` and is different from the **output** value, replace the **output** value with the **input** value
+* For a versioned release:
+  * If there is no **output** value, set the **output** value to an empty JSON array, convert the **input** value to a [versioned value](#versioned-values), and append it to the new array of versioned values in **output**
+  * If the **input** value is different from the value of the `value` field of the most recent versioned value in **output**, convert the **input** value to a [versioned value](#versioned-values), and append it to the array of versioned values in **output**
+
+#### Array values
+
+If the **input** array contains anything other than objects, treat the array as a literal value. Otherwise, there are two sub-routines for arrays of objects: whole list merge and identifier merge.
+
+##### Whole list merge
+
+An **input** array should be treated as a literal value if the corresponding field in a [dereferenced copy](https://jsonref.readthedocs.io/en/latest/) of the release schema has `"array"` in its `type` and if any of the following are also true:
```

Comment:
We could provide a compiled schema in OCDS for the convenience of reimplementations of merging written in something other than python.

There's some work happening in BODS to add generate a dereferenced schema as part of the standard build process which we might be able to re-use.

Reply:
I added a dereferenced release schema and linked to it in fb0aff6